packages/output-view/src/browser/output-view-widget.ts

```diff
@@ -0,0 +1,98 @@
+import { ObservableOutputChannel } from "@theia/languages/lib/common/observable-output-channel";
+import { ObservableOutputChannelManager } from "@theia/languages/lib/common";
+
+import "../../src/browser/style/output-view.css";
+
+export const OUTPUT_VIEW_KIND = 'outputView';
+
+@injectable()
+export class OutputViewWidget extends VirtualWidget {
+
+    private selectedChannel: ObservableOutputChannel;
+
+    constructor( @inject(ObservableOutputChannelManager) protected readonly outputChannelManager: ObservableOutputChannelManager) {
+        super();
+        this.id = OUTPUT_VIEW_KIND;
+        this.title.label = 'Output';
+        this.title.iconClass = 'fa fa-flag';
+        this.title.closable = true;
+        this.addClass('theia-output-view');
+        outputChannelManager.getChannels().forEach(this.registerListener.bind(this));
+        outputChannelManager.onChannelAdded(channel => {
+            this.registerListener(channel);
+            this.update();
+        });
+    }
+
+    protected registerListener(outputChannel: ObservableOutputChannel) {
+        if (!this.selectedChannel) {
+            this.selectedChannel = outputChannel;
+        }
+        outputChannel.onContentChange(c => {
+            if (outputChannel === this.selectedChannel) {
+                this.update();
+            }
+        });
+    }
+
+    render(): VirtualElement[] {
+        return [this.renderChannelSelector(), this.renderChannelContents()];
+    }
+
+    protected renderChannelContents(): VirtualElement {
+        if (this.selectedChannel) {
+            return h.div(
+                { id: 'outputContents' },
+                VirtualRenderer.flatten(this.selectedChannel.getLines().map(line => this.toHtmlText(line))));
+        } else {
+            return h.div({ id: 'outputContents' });
+        }
+    }
+
+    protected toHtmlText(text: string): VirtualElement[] {
+        const result: VirtualElement[] = [];
+        if (text) {
+            const lines = text.split(/([\n\r]+)/);
+            for (const line of lines) {
+                result.push(h.div(line));
+            }
+        }
+        return result;
+    }
+
+    protected renderChannelSelector(): VirtualElement {
+        const channelOptionElements: h.Child[] = [];
+        this.outputChannelManager.getChannelNames().forEach(channelName => {
+            channelOptionElements.push(h.option({ value: channelName }, channelName));
+        });
+
+        return h.select({
+            id: 'outputChannelList',
+            onchange: async event => {
+                const channelName = (event.target as HTMLSelectElement).value;
+                this.selectedChannel = this.outputChannelManager.getChannel(channelName);
+                this.update();
+            }
+        }, VirtualRenderer.flatten(channelOptionElements));
+    }
+
+    protected onUpdateRequest(msg: Message) {
+        super.onUpdateRequest(msg);
+        setTimeout(() => {
```

Comment:
`setTimeout` should not be necessary, this code is executed after rendering

Reply:
Unfortunately it is necessary.